pyenv/action.yml

```diff
@@ -0,0 +1,80 @@
+name: Install Pyenv Python
+description: Install and activate an interpreter via Pyenv.
+inputs:
+  python-version:
+    description: The version of Python to install.
+    required: true
+  pyenv-version:
+    description: The version of Pyenv to use. Should be a tag or branch name.
+    required: false
+    default: v2.3.21
+  pyenv-root:
+    description: Location for the PYENV_ROOT directory.
+    required: false
+    default: ${{ runner.tool_cache }}/pyenv_root
+outputs:
+  venv-dir:
+    description: The absolute path of the virtual environment directory created by this action.
+    value: ${{ steps.create-venv.outputs.venv-dir }}
+runs:
+  using: composite
+  steps:
+    - name: Cache Pyenv
+      uses: actions/cache@v3
+      with:
+        path: ${{ inputs.pyenv-root }}
+        key: ${{ matrix.os }}-${{ runner.arch }}-install-pyenv-${{ inputs.pyenv-version }}-${{ inputs.python-version }}-v1
+    - name: Install Pyenv
+      shell: bash
+      env:
+        PYENV_ROOT: ${{ inputs.pyenv-root }}
+      run: |
+        if [[ -d "${PYENV_ROOT}" ]]; then
+          echo "Using cached ${PYENV_ROOT}"
+        else
+          # We use a git fetch trick to avoid cloning - this relies on knowing the exact full sha we
```

Comment:
Could you explain a bit more about why avoiding cloning?

My understanding was 

```shell
git clone --depth 1 --branch ${{ input.pyenv-version }} https://github.com/pyenv/pyenv ${PYENV_ROOT}
```

would be equivalent to the sequence of `git` invocations here (despite the name, `--branch` is documented as working with tags).

Reply:
Running some tests right now, but, apparently, ignorance. Will fix depending on test outcome.

Reply:
You're correct. I used the 1st git commit of the linux kernel for a pretty heavyweight test. Both methods same size repo of course (~284MB) but your method faster since it has only 1 network round trip.

My method also works for shas with no refs and that is why it stuck in my head. Thanks for the prompt!

Reply:
Cool, no worries!

> My method also works for shas with no refs

Ah, yeah, I see that can be handy in some cases 👍